### ISSUE TITLE
CentOS: storaged crashes in poll_with_variant()

### ISSUE THREAD

**Issue by stefwalter** [cockpit-project/cockpit#5331]
https://github.com/storaged-project/storaged/pull/109

Should be fixed by above storaged pull request.

```
Nov 09 07:27:35 localhost.localdomain systemd-coredump[1920]: Process 1114 (storaged) of user 0 dumped core.
                                                              
                                                              Stack trace of thread 1114:
                                                              #0  0x00007fee4f184d1f poll_with_variant (libstoraged_lvm2.so)
                                                              #1  0x00007fee4f18534f variant_reader_watch_child (libstoraged_lvm2.so)
                                                              #2  0x00007fee6105e494 g_child_watch_dispatch (libglib-2.0.so.0)
                                                              #3  0x00007fee610617aa g_main_dispatch (libglib-2.0.so.0)
                                                              #4  0x00007fee61061af8 g_main_context_iterate (libglib-2.0.so.0)
                                                              #5  0x00007fee61061dca g_main_loop_run (libglib-2.0.so.0)
                                                              #6  0x0000000000414a5f main (storaged)
                                                              #7  0x00007fee60a5bb15 __libc_start_main (libc.so.6)
                                                              #8  0x0000000000414b9d _start (storaged)
```

**Comment by cockpituous** [comment]
verify/centos-7
Ooops, it happened again<hr>
```
Traceback (most recent call last):
  File "check-storage-lvm2", line 116, in testLvm
    "DISK2": True }})
  File "storagelib.py", line 220, in dialog_with_retry
    self.dialog_wait_close()
  File "storagelib.py", line 136, in dialog_wait_close
    self.browser.wait_not_present('#dialog')
  File "testlib.py", line 223, in wait_not_present
    return self.wait_js_func('!ph_is_present', selector)
  File "testlib.py", line 214, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "testlib.py", line 714, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "testlib.py", line 737, in _invoke
    raise Error(res['error'])
Error: timeout
Message did not receive a reply (timeout by message bus)
```
First occurrence: 2016-11-10T12:45:11.879313 | revision 204aab6a909647c2329848dcdc56789118784256, [logs](http://fedorapeople.org/groups/cockpit/logs/pull-5341-204aab6a-verify-centos-7/log.html)
Times recorded: 1
Latest occurrences:

- 2016-11-10T12:45:11.879313 | revision 204aab6a909647c2329848dcdc56789118784256, [logs](http://fedorapeople.org/groups/cockpit/logs/pull-5341-204aab6a-verify-centos-7/log.html)

**Comment by stefwalter** [comment]
CC @vpodzime

**Comment by cockpituous** [comment]
verify/centos-7
Ooops, it happened again<hr>
```
Traceback (most recent call last):
  File "check-storage-lvm2", line 155, in testLvm
    self.dialog({ "size": 30 })
  File "storagelib.py", line 251, in dialog
    self.dialog_wait_close()
  File "storagelib.py", line 236, in dialog_wait_close
    self.browser.wait_not_present('#dialog')
  File "testlib.py", line 221, in wait_not_present
    return self.wait_js_func('!ph_is_present', selector)
  File "testlib.py", line 212, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "testlib.py", line 740, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "testlib.py", line 766, in _invoke
    raise Error(res['error'])
Error: timeout
Message did not receive a reply (timeout by message bus)
```
First occurrence: 2017-04-11T15:17:39.453935 | revision 53548fc2173a41587238aa47f4b7e41e8efc258e, [logs](http://fedorapeople.org/groups/cockpit/logs/pull-6324-53548fc2-verify-centos-7/log.html)
Times recorded: 1
Latest occurrences:

- 2017-04-11T15:17:39.453935 | revision 53548fc2173a41587238aa47f4b7e41e8efc258e, [logs](http://fedorapeople.org/groups/cockpit/logs/pull-6324-53548fc2-verify-centos-7/log.html)

**Comment by cockpituous** [comment]
verify/centos-7
Ooops, it happened again<hr>
```
Traceback (most recent call last):
  File "/build/cockpit/bots/../test/verify/check-storage-lvm2", line 155, in testLvm
    self.dialog({ "size": 30 })
  File "/build/cockpit/test/verify/storagelib.py", line 251, in dialog
    self.dialog_wait_close()
  File "/build/cockpit/test/verify/storagelib.py", line 236, in dialog_wait_close
    self.browser.wait_not_present('#dialog')
  File "/build/cockpit/test/common/testlib.py", line 233, in wait_not_present
    return self.wait_js_func('!ph_is_present', selector)
  File "/build/cockpit/test/common/testlib.py", line 224, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "/build/cockpit/test/common/testlib.py", line 821, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "/build/cockpit/test/common/testlib.py", line 847, in _invoke
    raise Error(res['error'])
Error: timeout
Message did not receive a reply (timeout by message bus)
```
First occurrence: 2017-08-02T17:18:10.265075 | revision 6d3c995f8154cc380133ed4a6c9b8035c449341d, [logs](http://fedorapeople.org/groups/cockpit/logs/pull-7401-20170802-171010-6d3c995f-verify-centos-7/log.html)
Times recorded: 1
Latest occurrences:

- 2017-08-02T17:18:10.265075 | revision 6d3c995f8154cc380133ed4a6c9b8035c449341d, [logs](http://fedorapeople.org/groups/cockpit/logs/pull-7401-20170802-171010-6d3c995f-verify-centos-7/log.html)

**Comment by cockpituous** [comment]
verify/centos-7
Ooops, it happened again<hr>
```
# ----------------------------------------------------------------------
# testLvm (check_storage_lvm2.TestStorage)
#
Warning: Permanently added '[127.0.0.2]:2501' (ECDSA) to the list of known hosts.
> Using older 'storaged' API: org.storaged.Storaged
  WARNING: Sum of all thin volume sizes (100.00 MiB) exceeds the size of thin pool TEST1/pool and the size of whole volume group (48.00 MiB)!
> Message did not receive a reply (timeout by message bus)
Traceback (most recent call last):
  File "/build/cockpit/bots/../test/verify/check-storage-lvm2", line 118, in testLvm
    "DISK2": True }})
  File "/build/cockpit/test/verify/storagelib.py", line 327, in dialog_with_retry
    self.dialog_wait_close()
  File "/build/cockpit/test/verify/storagelib.py", line 247, in dialog_wait_close
    self.browser.wait_not_present('#dialog')
  File "/build/cockpit/test/common/testlib.py", line 233, in wait_not_present
    return self.wait_js_func('!ph_is_present', selector)
  File "/build/cockpit/test/common/testlib.py", line 224, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "/build/cockpit/test/common/testlib.py", line 825, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "/build/cockpit/test/common/testlib.py", line 851, in _invoke
    raise Error(res['error'])
Error: timeout
Message did not receive a reply (timeout by message bus)

not ok 90 testLvm (check_storage_lvm2.TestStorage) # duration: 126s
Wrote TestStorage-testLvm-centos-7-127.0.0.2-2501-FAIL.png
Wrote TestStorage-testLvm-centos-7-127.0.0.2-2501-FAIL.html
Wrote TestStorage-testLvm-centos-7-127.0.0.2-2501-FAIL.js.log
Journal extracted to TestStorage-testLvm-centos-7-127.0.0.2-2501-FAIL.log
Core dumps downloaded to /build/cockpit/TestStorage-testLvm-centos-7-127.0.0.2-2501-FAIL.core
```
First occurrence: 2017-10-26T09:16:31.383087 | revision da8f9e8f7f6fac4b5ec34a3c78278a4038073264, [logs](http://fedorapeople.org/groups/cockpit/logs/pull-7972-20171026-085959-da8f9e8f-verify-centos-7/log.html)
Times recorded: 1
Latest occurrences:

- 2017-10-26T09:16:31.383087 | revision da8f9e8f7f6fac4b5ec34a3c78278a4038073264, [logs](http://fedorapeople.org/groups/cockpit/logs/pull-7972-20171026-085959-da8f9e8f-verify-centos-7/log.html)

**Comment by cockpituous** [comment]
verify/centos-7
Ooops, it happened again<hr>
```
# ----------------------------------------------------------------------
# testLvm (check_storage_lvm2.TestStorage)
#
Warning: Permanently added '[127.0.0.2]:2201' (ECDSA) to the list of known hosts.
> Using older 'storaged' API: org.storaged.Storaged
  WARNING: Sum of all thin volume sizes (100.00 MiB) exceeds the size of thin pool TEST1/pool and the size of whole volume group (48.00 MiB)!
> Message did not receive a reply (timeout by message bus)
Traceback (most recent call last):
  File "/build/cockpit/bots/../test/verify/check-storage-lvm2", line 136, in testLvm
    "size": 20 })
  File "/build/cockpit/test/verify/storagelib.py", line 279, in dialog
    self.dialog_wait_close()
  File "/build/cockpit/test/verify/storagelib.py", line 257, in dialog_wait_close
    self.browser.wait_not_present('#dialog')
  File "/build/cockpit/test/common/testlib.py", line 236, in wait_not_present
    return self.wait_js_func('!ph_is_present', selector)
  File "/build/cockpit/test/common/testlib.py", line 227, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "/build/cockpit/test/common/testlib.py", line 790, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "/build/cockpit/test/common/testlib.py", line 816, in _invoke
    raise Error(res['error'])
Error: timeout
Message did not receive a reply (timeout by message bus)

not ok 50 testLvm (check_storage_lvm2.TestStorage) # duration: 117s
Wrote TestStorage-testLvm-centos-7-127.0.0.2-2201-FAIL.png
Wrote TestStorage-testLvm-centos-7-127.0.0.2-2201-FAIL.html
Wrote TestStorage-testLvm-centos-7-127.0.0.2-2201-FAIL.js.log
Journal extracted to TestStorage-testLvm-centos-7-127.0.0.2-2201-FAIL.log
Core dumps downloaded to /build/cockpit/TestStorage-testLvm-centos-7-127.0.0.2-2201-FAIL.core
```
First occurrence: 2017-12-01T11:54:27.584976 | revision 5743619882e3b30a75028305ff5996ad0ce80158, [logs](http://fedorapeople.org/groups/cockpit/logs/pull-8087-20171201-112121-57436198-verify-centos-7/log.html)
Times recorded: 1
Latest occurrences:

- 2017-12-01T11:54:27.584976 | revision 5743619882e3b30a75028305ff5996ad0ce80158, [logs](http://fedorapeople.org/groups/cockpit/logs/pull-8087-20171201-112121-57436198-verify-centos-7/log.html)

**Comment by martinpitt** [comment]
I'm not convinced that this is fixed. The last storaged fix in RHEL 7.4 (which should transcend to CentOS) was in June 2017, and it was not related to this crash. So let's keep this open for another round or two.

**Comment by martinpitt** [comment]
*bump* again; this happens seldomly, but http://pkgs.devel.redhat.com/cgit/rpms/storaged/log/?h=extras-rhel-7.4 has no new commits since.
